samples/csharp_dotnetcore/80.skills-simple-bot-to-bot/SimpleRootBot/AdapterWithErrorHandler.cs

```diff
@@ -17,57 +18,96 @@ namespace Microsoft.BotBuilderSamples.SimpleRootBot
-                        // a chance to clean up.
-                        // Note: "activeSkillProperty" is set by the RooBot while messages are being
-                        // forwarded to a Skill.
-                        var activeSkill = await conversationState.CreateProperty<BotFrameworkSkill>("activeSkillProperty").GetAsync(turnContext, () => null);
-                        if (skillClient != null && skillsConfig != null && activeSkill != null)
-                        {
-                            var botId = configuration.GetSection(MicrosoftAppCredentials.MicrosoftAppIdKey)?.Value;
+            await sendErrorMessageAsync(turnContext, exception);
+            await endSkillConversationAsync(turnContext);
+            await clearConversationStateAsync(turnContext);
+        }
 
-                            var endOfConversation = Activity.CreateEndOfConversationActivity();
-                            endOfConversation.Code = EndOfConversationCodes.Unknown;
-                            endOfConversation.ApplyConversationReference(turnContext.Activity.GetConversationReference(), true);
+        protected async Task sendErrorMessageAsync(ITurnContext turnContext, Exception exception)
+        {
+            // Send a message to the user
+            var errorMessageText = "The bot encountered an error or bug.";
+            var errorMessage = MessageFactory.Text(errorMessageText, errorMessageText, InputHints.IgnoringInput);
+            await turnContext.SendActivityAsync(errorMessage);
 
-                            await skillClient.PostActivityAsync(botId, activeSkill, skillsConfig.SkillHostEndpoint, (Activity)endOfConversation, CancellationToken.None);
-                        }
+            errorMessageText = "To continue to run this bot, please fix the bot source code.";
+            errorMessage = MessageFactory.Text(errorMessageText, errorMessageText, InputHints.ExpectingInput);
+            await turnContext.SendActivityAsync(errorMessage);
 
-                        // Delete the conversationState for the current conversation to prevent the
-                        // bot from getting stuck in a error-loop caused by being in a bad state.
-                        // ConversationState should be thought of as similar to "cookie-state" in a Web pages.
-                        await conversationState.DeleteAsync(turnContext);
-                    }
-                    catch (Exception ex)
-                    {
-                        logger.LogError(ex, $"Exception caught on attempting to Delete ConversationState : {ex}");
-                    }
-                }
+            // Send a trace activity, which will be displayed in the Bot Framework Emulator
+            await turnContext.TraceActivityAsync("OnTurnError Trace", exception.ToString(), "https://www.botframework.com/schemas/error", "TurnError");
+        }
+
+        private async Task<InvokeResponse> endSkillConversationAsync(ITurnContext turnContext)
```

Comment:
We don't seem to use the InvokeResponse value at all, this method should probably be `private async Task` so we don't return nulls or anything else that we don't use.